### PR TITLE
Feature/update scala

### PR DESCRIPTION
**Pull request by Grifs**

### CODE REVIEW

File: src/main/scala/io/viash/ViashNamespace.scala [hunk 1]
```diff
@@ -303,12 +304,12 @@ object ViashNamespace {
...
     }
 
-    for (data <- if (parallel) collectedData.par else collectedData) {
+    for (data <- if (parallel) collectedData.par.iterator else collectedData) {
       // remove trailing + or ; mode character
       val commandNoMode = command.replaceFirst(""" \\?[;+]$""", "")
       val replacedCommand = 
         fields.foldRight(commandNoMode){ (field, command) => 
-          command.replaceAllLiterally(s"{$field}", data.getField(field).get)
+          command.replace(s"{$field}", data.getField(field).get)
```
Comment (rcannood) [inline src/main/scala/io/viash/ViashNamespace.scala:312]: Will this still replace all occurrences of `{field}`?

File: src/main/scala/io/viash/functionality/arguments/package.scala [hunk 2]
```diff
@@ -88,7 +88,7 @@ package object arguments {
 
   implicit def encodeArgument[A <: Argument[_]]: Encoder[A] = Encoder.instance {
     par =>
-      val typeJson = Json.obj("type" → Json.fromString(par.`type`))
+      val typeJson = Json.obj("type" -> Json.fromString(par.`type`))
```
Comment (rcannood) [inline src/main/scala/io/viash/functionality/arguments/package.scala:91]: RIP `→` :(

**Review by rcannood — COMMENTED** [review]
See question

File: src/main/scala/io/viash/ViashNamespace.scala (hunk 1 above)
Comment (Grifs) [reply src/main/scala/io/viash/ViashNamespace.scala:312]:
```

File: src/main/scala/io/viash/functionality/arguments/package.scala (hunk 2 above)
Comment (Grifs) [reply src/main/scala/io/viash/functionality/arguments/package.scala:91]: * sad slide whistle *

**Review by rcannood — APPROVED** [review]
LGTM! Please merge :)

**Pull request merged**